client/engine.go

```diff
@@ -145,16 +145,14 @@ func performOperation(opType int, ev termbox.Event, conn *websocket.Conn) {
 	case OperationInsert:
 		logger.Infof("LOCAL INSERT: %s at cursor position %v\n", ch, e.Cursor)
 
-		r := []rune(ch)
-		e.AddRune(r[0])
-
-		text, err := doc.Insert(e.Cursor, ch)
+		text, err := doc.Insert(e.Cursor+1, ch)
```

Comment:
question: why is this now using `e.Cursor+1`?

Reply:
Before, the `AddRune` method was inserting a rune into the editor (not CRDT) text and then moving the cursor with `MoveCursor(1, 0)`. Immediately afterwards, this `doc.Insert` happened, which overwrote the changes to the editor text that `AddRune` made, but used the new cursor position for the insertion.

Once I removed AddRune, I had to account for that previous cursor movement to get the same behavior. I landed on this mostly through trial and error, to be honest. I tried out some different combinations and orderings for moving the cursor and inserting, and this was the best combination of correct+simple that I found. I think the `e.Cursor+1` accounts for the actual behavior of `doc.Insert`.

You're definitely welcome to play around with it yourself. There could be a more obvious solution that I missed.

Reply:
Whoops! I somehow missed this comment; sorry for the delay. I tested it out and it looks fine!

And you're correct with the `e.Cursor+1` approach; it makes sense to have it that way.